pddl/logic/base.py

```diff
@@ -209,6 +214,69 @@ class Not(UnaryOp):
 
     SYMBOL = "not"
 
+@cache_hash
+@functools.total_ordering
+class QuantifiedCondition(Formula):
+    """Superclass for quantified conditions."""
+
+    SYMBOL: str
+
+    def __init__(
+        self, cond: "Formula", variables: Optional[Collection[Variable]] = None
+    ) -> None:
+        """Initialize the quantified condition."""
+        self._cond = cond
+        self._variables = ensure_set(variables)
+
+    @property
+    def condition(self) -> "Formula":
+        """Get the condition."""
+        return self._cond
+
+    @property
+    def variables(self) -> AbstractSet[Variable]:
+        """Get the variables."""
+        return self._variables
+
+    def __str__(self) -> str:
+        """Get the string representation."""
+        var_block = ' '.join([f'{v} - {" ".join(v.type_tags)}' for v in self.variables])
```

Comment:
Are variables always typed here?

Reply:
Exactly. If there are no types, then `-` will be included anyway. It is a set because, as you said, they can have multiple types